File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.py
class Solution:
    def missingMultiple(self, nums: List[int], k: int) -> int:
        seen = set()
        for x in nums:
            if x % k == 0:
                seen.add(x // k)
        i = 1
        while i in seen:
            i += 1
        return i * k

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.rs
use std::collections::HashSet;

impl Solution {
    pub fn missing_multiple(nums: Vec<i32>, k: i32) -> i32 {
        let mut seen = HashSet::new();
        for &x in &nums {
            if x % k == 0 {
                seen.insert(x / k);
            }
        }
        let mut i = 1;
        while seen.contains(&i) {
            i += 1;
        }
        i * k
    }
}